.github/workflows/rust.yml

```diff
@@ -10,13 +10,34 @@ env:
   CARGO_TERM_COLOR: always
 
 jobs:
+
   build:
+    runs-on: ${{ matrix.os }}
 
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
```

Comment:
You might not need to do this on multiple OS'es - consider utilizing https://rust-lang.github.io/rustup/cross-compilation.html.

Reply:
https://users.rust-lang.org/t/how-to-build-for-all-operating-systems/77932/3

There are some concerns with just using the cargo target system, but the post above also linked to a crate that handles VMs internally. It only requires the runners to have podman or docker, which the ubuntu-latest does. 

https://github.com/cross-rs/cross?tab=readme-ov-file#supported-targets
https://github.com/cross-rs/cross-toolchains